README.md

```diff
@@ -40,364 +43,346 @@ some kind of discriminant:
+proposal are intended to align with that of TypeScript's `enum` where possible.
 
-| Specification Name | \[\[Description]] | Value and Purpose |
-|:-|:-|:-|
-| @@toEnum | `"Symbol.toEnum"` | A method that is used to derive the value for an enum member during _EnumMember_ evaluation. |
-| @@formatEnum | `"Symbol.formatEnum"` | A method of an _enum object_ that is used to convert a value into a string representation based on the member names of the enum. Called by `Enum.format`. |
-| @@parseEnum | `"Symbol.parseEnum"` | A method of an _enum object_ that is used to convert a member name String into the value represented by that member of the enum. Called by `Enum.parse`. |
 
 ## Enum Declarations
 
 Enum declarations consist of a finite set of _enum members_ that define the names and values
 for each member of the enum. These results are stored as properties of an _enum object_. An 
-_enum object_ is an ordinary object with an \[\[EnumMembers]] internal slot, and whose 
-\[\[Prototype]] is `null`.
+_enum object_ is an ordinary object whose \[\[Prototype]] is `null`. Each enum member defines
+a property on the _enum object_.
 
-### Automatic Initialization
+In addition, an _enum object_ contains an `@@iterator` method that yields a `[key, value]` entry for each declared enum
+member, in document order. To explain the semantics of the `@@iterator` method, an _enum object_ may require an
+\[\[EnumMembers]] internal slot.
 
-If an _enum member_ does not supply an _Initializer_, the value of that _enum member_ will be 
-automatically initialized:
 
-```js
-enum DaysOfTheWeek {
-  Sunday, // 0
-  Monday, // 1
-  Tuesday, // 2
-  // etc.
-}
-```
+## Enum Members
+
+Enum members define the set of values that belong to the enum's domain. Each enum member consists of a name and an
+initializer that defines the value associated with that name. Enum members are \[\[Writable]]: `false` and
+\[\[Configurable]]: `false`.
+
+
+### Enum Member Names
+
+Enum member names are currently restricted to _IdentifierName_ and _StringLiteral_, as those are the only member names
+currently supported by TypeScript's `enum`. We may opt to expand this to allow other member names like
+_ComputedPropertyName_ in the future, if there is sufficient motivation.
+
+An enum may not have duplicate member names. We may opt to introduce restrictions on member names like `constructor` if
+we deem it necessary to support ADT enums in the future.
+
+ If an enum member name shares the same string value as the name of the enum itself, it shadows the enum declaration
+ name within the enum body.
+
+### Enum Member Initializers
+
+An enum member's initializer is restricted to a subset of
+ECMAScript values (i.e., `Number`, `String`, `Symbol`, and `BigInt`). This limitation allows us to consider future
+support for Algebraic Data Types (ADT) in enums without the possibility of conflicting with something like an enum
+member whose value is a function.
```

Comment:
At first I didn't understand that "an enum member whose value is a function" is exactly what an ADT would consist of, so it would help to connect the two more explicitly.